modules/router-store/src/serializer.ts

```diff
@@ -0,0 +1,15 @@
+import { RouterStateSnapshot } from '@angular/router';
+
+export type RouterStateSnapshotType =
+  | RouterStateSnapshot
+  | Partial<RouterStateSnapshot>;
+
+export abstract class RouterStateSerializer {
+  abstract serialize(routerState: RouterStateSnapshotType | null): RouterStateSnapshotType | null;
```

Comment:
Maybe we guard against nulls outside? Cause I think it should never be called with `null`.